src/js/angular/clustermanagement/directives/cluster-configuration/multi-region.directive.js

```diff
@@ -15,7 +23,197 @@ function MultiRegion() {
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.secondary'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.disableSecondaryMode = () => {
+                $scope.setLoader(true);
+                ModalService.openSimpleModal({
+                    title: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.disable_secondary_mode'),
+                    message: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.disable_secondary_mode_warning'),
+                    warning: true,
+                    backdrop: 'static',
+                    stopPropagation: true
+                }).result
+                    .then(() => {
+                        $scope.setLoader(true);
+                        return ClusterRestService.disableSecondaryMode();
+                    })
+                    .then(() => toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.disabled_secondary_mode')))
+                    .catch((response) => {
+                        if (response === 'cancel') {
+                            return;
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.disabling'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.setLoader = (loader, message) => {
+                $timeout.cancel($scope.loaderTimeout);
+                if (loader) {
+                    $scope.loaderMessage = message;
+                    $scope.loaderTimeout = $timeout(() => {
+                        $scope.loader = loader;
+                    }, 50);
+                } else {
+                    $scope.loader = false;
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+            const setTopology = (clusterModel) => {
+                const leader = clusterModel.nodes.find((node) => node.nodeState === NodeState.LEADER);
+                $scope.topology = leader.topologyStatus;
+            };
+
+            // =========================
+            // Events and watchers
+            // =========================
+            const subscribeHandlers = () => {
+                subscriptions.push($scope.$watch('clusterModel', function(newValue, oldValue) {
+                    if (newValue !== oldValue) {
+                        ClusterRestService.getClusterConfig()
+                            .then((response) => {
+                                $scope.secondaryTag = response.data.secondaryTag;
```

Comment:
Also, at some point we need to create models and do some transformation so that components to work with the internal model and not with the response from the backend

Reply:
extended the existing model